game/src/core/components/DrawComponent.java

```diff
@@ -216,6 +203,21 @@ public void queueAnimation(int forFrames, final IPath... next) {
     }
   }
 
+  /**
+   * Queue up an Animation to be considered as the next played Animation.
+   *
+   * <p>Animations are given as an {@link IPath} Array or multiple variables. Animation length is
+   * set to the number of frames given for an animation. If you need to queue longer Animations, use
+   * {@link #queueAnimation(int, IPath...)}. The first existing Animation will be added to the
+   * queue.
+   *
+   * @param next Array of IPaths representing the Animation.
+   */
+  public void queueAnimation(final IPath... next) {
+    Arrays.stream(next)
+        .forEach(path -> animation(path).ifPresent(anim -> queueAnimation(anim.duration(), path)));
+  }
```

Comment:
Das mit der Frametime ergibt in meinen Augen Sinn (ohne nochmal richtig tief eingestiegen zu sein).
Aber diese Funktion ist jetzt nicht mehr Semantisch identisch zum alten verhalten.
Du fügst jetzt ALLE Animationen in `next` hinzu, ` queueAnimation(int forFrames, final IPath... next)` fügt aber nur die eine (und zwar die erste existierende Animation) hinzu. 

```java
   * <p>Animations are given as an {@link IPath} Array or multiple variables. The First existing
   * Animation will be added to the queue. If the Animation is already added, the remaining Frames
   * are set to the highest of remaining or new.
 ```

Reply:
Das war doch vorher auch so: Da wurde das variadische Argument `next` zusammen mit dem Faktor `1` an die "große" Methode reingereicht, die dann mit einer `for`-Schleife über alle Elemente im variadischen Parameter iteriert ... D.h. da wurde auch alles in `next` eingetragen :)

Aber es gibt trotzdem einen Unterschied: In der alten Methode wurde **alles*  in `next` weitergereicht, in der neuen Variante nur, wenn es für den Pfad eine Animation gibt. (Allerdings wird in der "großen" Methode in der `for`-Schleife dann auch wieder geschaut, ob sich zu dem aktuellen Pfad eine Animation finden lässt und nur dann wird sie in die Queue gepackt - der Unterschied ist also nur, dass das Aussieben jetzt bereits hier in der "kleinen" Methode stattfindet.)

Reply:
@AMatutat Imho ist der Javadoc einfach falsch :)

Reply:
you are correct......

Hmm... ich erinnere mich an lange Diskussionen über die Funktion....
Dann fix mal die javadoc gleich mit und dann lgtm

Reply:
Hmmm. Wenn ich mir die Stellen anschaue, wo das genutzt wird, dann wird i.d.R. tatsächlich auch nur eine Animation übergeben. Geht halt auch mit variadisch. Die Frage ist nur: Wollen wir das so, gerade unter dem Blickwinkel, was Du oben gesagt hattest?

Reply:
> Edit: Ein Teil der Funktionsweise sollte (gewesen) sein, dass ich beim Queuen der Animation semantisch so etwas machen kann wie: "Pack Animation A rein, und wenn es die nicht gibt, dann Animation B, und wenn es die nicht gibt, dann C ...."
> 
> Das war ein häufig aufkommendes Szenario.
> 
> Jetzt bin ich verwirrt, dass wir damals dann doch alle Animationen in die Queue gepackt haben. Ich vermute gerade, dass durch die identisch definierte Framezeit eh nur die erste Animation SICHTBAR abgespielt worden wäre.... Müsste das aber mit den Prioritäten verheiraten.... Nee... macht auch keinen Sinn.

Naja, in der "großen" Methode ist das hier drin:

```java
  public void queueAnimation(int forFrames, final IPath... next) {
    for (IPath path : next) {
      // is an existing animation of the component
      if (animationMap.containsKey(path.pathString())) {
        // check if the path is already queued
        if (animationQueue.containsKey(path)) {
          // update time of the animation
          animationQueue.put(path, Math.max(animationQueue.get(path), forFrames));
        } else {
          // add animation
          animationQueue.put(path, forFrames);
        }
        return;
      }
    }
  }
```

- Gehe über alle Pfade/Animationen
    - wenn diese Animation existiert (d.h. in der `animationMap` ein Eintrag ist)
        - ist das Ding schon in der Queue => pack sie nochmal rein, aber mit der neuen Dauer `forFrames`
        - ist das ding nicht in der Queue => pack sie einfach neu rein mit der Dauer `forFrames`

Reply:
Argl. Du hast recht: Die erste gefundene Animation wird hinzugefügt - und dann passiert das stille `return` ... D.h. die äußere `for`-loop bricht dann ab, das habe ich übersehen.

Aber es ändert sich in Bezug auf die "kleine" Methode auch nix: Diese filtert bereits alle nicht vorhandenen Animationspfade raus und übergibt den Rest an die "große" Methode, und die bricht nach dem ersten Hinzufügen ab.

Der einzige Unterschied ist die Default-Anzeigedauer: Früher 1 Frame, jetzt halt das, was die Animation für einen kompletten Durchlauf braucht (vermutlich 5 Frames in der Größenordnung).

Reply:
Hmmm. So richtig zufrieden bin ich noch nicht. In dieser Methode, die ja eigentlich nur eine Convenience-Methode für die Kunden ist, machen wir relativ ähnlichen Code wie in der "großen" Hauptmethode - dieses ganze Abfragen, ob irgendwas da ist oder nicht, ist doppelt. Da stimmt von der Architektur her noch was nicht.

Reply:
> Argl. Du hast recht: Die erste gefundene Animation wird hinzugefügt - und dann passiert das stille return ... D.h. die äußere for-loop bricht dann ab, das habe ich übersehen.

Bäh.... das übersieht man an der stelle aber wirklich schnell

Reply:
> > Edit: Ein Teil der Funktionsweise sollte (gewesen) sein, dass ich beim Queuen der Animation semantisch so etwas machen kann wie: "Pack Animation A rein, und wenn es die nicht gibt, dann Animation B, und wenn es die nicht gibt, dann C ...."
> > Das war ein häufig aufkommendes Szenario.
> > Jetzt bin ich verwirrt, dass wir damals dann doch alle Animationen in die Queue gepackt haben. Ich vermute gerade, dass durch die identisch definierte Framezeit eh nur die erste Animation SICHTBAR abgespielt worden wäre.... Müsste das aber mit den Prioritäten verheiraten.... Nee... macht auch keinen Sinn.
> 
> Naja, in der "großen" Methode ist das hier drin:
> 
> ```java
>   public void queueAnimation(int forFrames, final IPath... next) {
>     for (IPath path : next) {
>       // is an existing animation of the component
>       if (animationMap.containsKey(path.pathString())) {
>         // check if the path is already queued
>         if (animationQueue.containsKey(path)) {
>           // update time of the animation
>           animationQueue.put(path, Math.max(animationQueue.get(path), forFrames));
>         } else {
>           // add animation
>           animationQueue.put(path, forFrames);
>         }
>         return;
>       }
>     }
>   }
> ```
> 
>  * Gehe über alle Pfade/Animationen
>   
>   * wenn diese Animation existiert (d.h. in der `animationMap` ein Eintrag ist)
>     
>     * ist das Ding schon in der Queue => pack sie nochmal rein, aber mit der neuen Dauer `forFrames`
>     * ist das ding nicht in der Queue => pack sie einfach neu rein mit der Dauer `forFrames`

@AMatutat Inversion of Control ist das Zauberwort. Ich habe es geschafft, dass der obige Ablauf 1:1 in folgender Zeile enthalten ist 

```java
Arrays.stream(next).filter(this::hasAnimation).findFirst().ifPresent(fn);
```

:)

Reply:
OK, `fn` muss man dann spezifisch definieren. Insbesondere ist dort der Unterschied zwischen den beiden `queueAnimation`-Funktionen zu definieren (vorgegebene Dauer vs. Anzahl der vorhandenen Animationsschritte) ...